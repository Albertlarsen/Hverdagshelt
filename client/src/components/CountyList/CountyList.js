import {Col, Button, Form, FormGroup, Label, Grid} from 'react-bootstrap';
import {CountyService} from "../../services";
import {Component} from 'react';
import * as React from 'react';
import {Alert} from "../../widgets";
import ReactDOM from 'react-dom';
//import {County} from "../../classTypes";
import Select from "react-select";
import Row from "react-bootstrap/es/Row";

let countyService = new CountyService();


export class CountyList extends Component<Props, State> {

  constructor(props) {
    super(props);
    this.state = {
      isLoaded: false,
      choosen: {label: "Bergen", countyId: 1},
      values: [
        {label: "Bergen", countyId: 1}
      ]
    };

    this.handleChangeCounty = this.handleChangeCounty.bind(this)
  }


  handleChangeCounty(e: Object) {

    this.props.handleOnChangeCounty(JSON.parse(e.value));

  };

  componentWillMount() {
    var arr = [];
    countyService
      .getCounties()
      .then(county2 => {
        county2.map(e => {
          var elem = {
            name: e.name,
            countyId: e.countyId
          };
          arr = arr.concat(elem)

        });
        this.setState({
          values: arr
        })
      })
      .catch((error: Error) => Alert.danger(error.message))

  }

  render() {
    let optionTemplate = this.state.values.map(v => {
      var data = {label: v.name, value: v.countyId, countyId: v.countyId};
      return (data)
    });
    return (
          <Select
            placeholder={'Hjemmekommune'}
            name="colors"
            options={optionTemplate}
            className="basic"
            classNamePrefix="select"
            onChange={this.handleChangeCounty}
          />
    );
  }
}