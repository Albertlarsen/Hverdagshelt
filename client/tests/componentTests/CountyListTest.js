// @flow

import {shallow} from "enzyme/build";
import * as React from "react";
import {CountyList} from "../../src/components/CountyList/CountyList";

describe('Test for CountyList component', () => {
    const wrapper = shallow(<CountyList />);


    it('renders correctly', () => {
        expect(wrapper).toMatchSnapshot();
    });

});