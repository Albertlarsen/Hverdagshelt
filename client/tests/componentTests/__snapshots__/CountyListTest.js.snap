// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test for CountyList component renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CountyList />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "className": "basic",
      "classNamePrefix": "select",
      "defaultInputValue": "",
      "defaultMenuIsOpen": false,
      "defaultValue": null,
      "name": "colors",
      "onChange": [Function],
      "options": Array [
        Object {
          "countyId": 1,
          "label": undefined,
          "value": 1,
        },
      ],
      "placeholder": "Hjemmekommune",
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "className": "basic",
        "classNamePrefix": "select",
        "defaultInputValue": "",
        "defaultMenuIsOpen": false,
        "defaultValue": null,
        "name": "colors",
        "onChange": [Function],
        "options": Array [
          Object {
            "countyId": 1,
            "label": undefined,
            "value": 1,
          },
        ],
        "placeholder": "Hjemmekommune",
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
